objects/src/accounts/package/mod.rs

```diff
@@ -0,0 +1,396 @@
+            for key in storage_entry.template_keys() {
+                key_set.insert(key.clone());
+            }
+        }
+        key_set
+    }
+
+    pub fn name(&self) -> &str {
+        &self.name
+    }
+
+    pub fn description(&self) -> &str {
+        &self.description
+    }
+
+    pub fn version(&self) -> &Version {
+        &self.version
+    }
+
+    pub fn targets(&self) -> &BTreeSet<AccountType> {
+        &self.targets
+    }
+
+    pub fn storage_entries(&self) -> &Vec<StorageEntry> {
+        &self.storage
+    }
+}
+
+#[derive(Debug, Error)]
+pub enum ComponentPackageError {
+    #[error("error creating AccountComponent")]
+    AccountComponentError(#[source] AccountError),
+    #[error("error trying to deserialize from toml")]
+    DeserializationError(#[from] toml::de::Error),
+    #[error("slot {0} is defined multiple times")]
+    DuplicateSlot(u8),
+    #[error("component storage slots have to start at 0")]
+    IncorrectStorageFirstSlot,
+    #[error("template value was not of the expected type {0}")]
+    IncorrectTemplateValue(String),
+    #[error("multi-slot entry should contain as many values as storage slots indices")]
+    InvalidMultiSlotEntry,
+    #[error("error deserializing component metadata: {0}")]
+    MetadataDeserializationError(String),
+    #[error("component storage slots are not contiguous")]
+    NonContiguousSlots,
+    #[error("error creating storage map")]
+    StorageMapError(#[source] AccountError),
+    #[error("template value ({0}) was not provided in the map")]
+    TemplateValueNotProvided(String),
+}
+
+// SERIALIZATION
+// ================================================================================================
+
+impl Serialize for AccountType {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let s = match self {
+            AccountType::FungibleFaucet => "FungibleFaucet",
+            AccountType::NonFungibleFaucet => "NonFungibleFaucet",
+            AccountType::RegularAccountImmutableCode => "RegularAccountImmutableCode",
+            AccountType::RegularAccountUpdatableCode => "RegularAccountUpdatableCode",
+        };
+        serializer.serialize_str(s)
+    }
+}
+
+impl<'de> Deserialize<'de> for AccountType {
+    fn deserialize<D>(deserializer: D) -> Result<AccountType, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let s: String = Deserialize::deserialize(deserializer)?;
+
+        match s.to_lowercase().as_str() {
+            "fungiblefaucet" => Ok(AccountType::FungibleFaucet),
+            "nonfungiblefaucet" => Ok(AccountType::NonFungibleFaucet),
+            "regularaccountimmutablecode" => Ok(AccountType::RegularAccountImmutableCode),
+            "regularaccountupdatablecode" => Ok(AccountType::RegularAccountUpdatableCode),
+            other => Err(D::Error::invalid_value(Unexpected::Str(other), &"a valid account type")),
+        }
```

Comment:
Should we make this deserialization more strict? (i.e., not do `to_lowercase()`)

Reply:
Changed this